src/Sparrow.Server/Platform/MemoryLockUsage.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Sparrow.Binary;
+using Sparrow.Global;
+using Sparrow.Platform;
+
+namespace Sparrow.Server.Platform;
+
+public static unsafe class MemoryLockUsage
+{
+        public static long LockedMemorySize;
+        
+        public static event EventHandler<long> MemoryLockedCalled;
+
+        public static void UpdateLockedMemory(Int64 v, char* filenamePtr)
+        {
+            MemoryLockedCalled?.Invoke(null, v);
+            Interlocked.Add(ref LockedMemorySize, v);
+        }
+        
+        public static bool RecoverLockedMemoryFailure(Int64 sizeToLock, char* filenamePtr)
+        {
+            if (PlatformDetails.RunningOnPosix)
+                return false; // nothing to do here
```

Comment:
Can you explain why there is nothing do to on Posix?

Reply:
There is no error recovery from locked memory in Linux.
In Windows, we can try to increase the working set size and retry, but on Linux, if it fails, that is it.